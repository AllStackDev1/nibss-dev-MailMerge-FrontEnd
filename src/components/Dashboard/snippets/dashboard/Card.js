import React from 'react'
import { Card } from 'styles/styled-components/DashboardCards'
import Loader from 'styles/styled-components/Loader';

const DashboardCard = ({ documents, image, key, label }) => (
    <Card className="padding-20 border-box">
        <div className="full-width display-flex space-between align-items-center">
            <div className="width-50 height-50 border-radius-100-percent display-flex align-items-center justify-center">
                <img src={image} className="height-20" alt="NIBSS Empty" />
            </div>
            {documents?.documents && documents.fetching !== true ?
                <span className="bold size-two-rem">{documents?.documents?.[key] || 0}</span>
                :
                <Loader className="lds-ring"><div></div><div></div><div></div><div></div></Loader>}
        </div>
        <p className="absolute bottom-20 left-20 bold size-pointeight-rem">{label}</p>
    </Card>
)

export default DashboardCard;
