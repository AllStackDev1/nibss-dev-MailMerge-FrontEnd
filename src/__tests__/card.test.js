import React from 'react';
import { shallow } from 'enzyme';
import Card from "../components/Dashboard/snippets/dashboard/Card"



const shallowSetup = (props = {}) => {

    return shallow(<Card {...props} />)
}

it("should test conditions", () => {

    const wrapper = shallowSetup({ documents: { documents: {}, fetching: false } })
    expect(wrapper).toMatchSnapshot()

})