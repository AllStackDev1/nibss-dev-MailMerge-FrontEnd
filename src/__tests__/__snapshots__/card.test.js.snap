// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test conditions 1`] = `ShallowWrapper {}`;
